cli/chaosli/cmd/context.go

```diff
@@ -0,0 +1,388 @@
+	targets, err := exec.Command("bash", "-c", cmd).Output()
+
+	if err != nil {
+		return nil, fmt.Errorf("could not grab list of targets names correlating to target selector: %v", err)
+	}
+
+	s.Stop()
+
+	targetsShow := []string{}
+	targetsAll := []string{}
+	targetsSplit := strings.Split(string(targets), "\n")
+
+	for i := 0; i < len(targetsSplit); i++ {
+		if len(targetsShow) < MAXTARGETSHOW {
+			targetsShow = append(targetsShow, targetsSplit[i])
+		}
+
+		if targetsSplit[i] == "NAME" || targetsSplit[i] == "" {
+			continue
+		}
+
+		targetsAll = append(targetsAll, targetsSplit[i])
+	}
+
+	fmt.Printf("\n🎯 There are %d %ss that will be targeted\n", size, level)
+
+	if size >= MAXTARGETSHOW {
+		fmt.Println("📝 Here are a small set of those targets:")
+	}
+
+	for _, target := range targetsShow {
+		fmt.Println(target)
+	}
+
+	if size >= MAXTARGETSHOW {
+		fmt.Println("...")
+	}
+
+	PrintSeparator()
+
+	return targetsAll, nil
+}
+
+func getTargetSize(disruption v1beta1.Disruption) (int, error) {
+	level := disruption.Spec.Level
+	podNamespace := disruption.Namespace
+	labels := disruption.Labels
+	cmd := fmt.Sprintf("kubectl get %s -n %s -l %s | wc -l", level, podNamespace, labels)
+	sizeString, err := exec.Command("bash", "-c", cmd).Output()
+
+	if err != nil {
+		return -1, fmt.Errorf("could not count the number of targets correlating to target selector: %v", err)
+	}
+
+	size, err := strconv.Atoi(strings.Trim(string(sizeString), "\n"))
+
+	if err != nil {
+		return -1, fmt.Errorf("could not convert string to integer in context target size: %v", err)
+	}
+
+	return size, nil
+}
+
+func grabDataForTargets(targets []string, disruption v1beta1.Disruption) ([]v1.Pod, []v1.Node, error) {
+	namespace := disruption.ObjectMeta.Namespace
+	level := disruption.Spec.Level
+	pods := []v1.Pod{}
+	nodes := []v1.Node{}
+
+	s := spinner.New(spinner.CharSets[38], 100*time.Millisecond)
+	s.Start()
+
+	for _, targetName := range targets {
+		pod := v1.Pod{}
+		node := v1.Node{}
+
+		cmd := fmt.Sprintf("kubectl get %s -o json -n %s %s", level, namespace, targetName)
+		targetData, err := exec.Command("bash", "-c", cmd).Output()
+
+		if err != nil {
+			return nil, nil, fmt.Errorf("could not grab target data: %v", err)
+		}
+
+		if level == types.DisruptionLevelPod {
+			if err := json.Unmarshal(targetData, &pod); err != nil {
+				return nil, nil, fmt.Errorf("json encoding failed: %v", err)
+			}
+
+			pods = append(pods, pod)
+		} else {
+			if err := json.Unmarshal(targetData, &node); err != nil {
+				return nil, nil, fmt.Errorf("json encoding failed: %v", err)
+			}
+			nodes = append(nodes, node)
+		}
+	}
+
+	s.Stop()
+
+	return pods, nodes, nil
+}
+
+func printContainerStatus(targetInfo []v1.Pod) {
+	percentCollect := make(map[string]float64)
+
+	fmt.Println("\nLets take" +
+		" a look at the status of your Targeted Pod Containers...")
+
+	totalContainers := 0
+
+	for i := 0; i < len(targetInfo); i++ {
```

Comment:
I think both these for loops can be shrunken slightly by using the `for idx, element := range $list` syntax